haystack/nodes/audio/_text_to_speech.py

```diff
@@ -33,17 +34,22 @@ def __init__(
         model_name_or_path: Union[str, Path],
         use_gpu: bool = True,
         transformers_params: Optional[Dict[str, Any]] = None,
+        devices: Optional[List[Union[str, torch.device]]] = None,
     ):
         """
         :param model_name_or_path: The text to speech model, for example `espnet/kan-bayashi_ljspeech_vits`.
         :param use_gpu: Whether to use GPU (if available). Defaults to True.
         :param transformers_params: Parameters to pass over to the `Text2Speech.from_pretrained()` call.
+        :param devices: List of torch devices (e.g. cuda, cpu, mps) to limit inference to specific devices.
+                        A list containing torch device objects and/or strings is supported (For example
+                        [torch.device('cuda:0'), "mps", "cuda:1"]). When specifying `use_gpu=False` the devices
+                        parameter is not used and a single cpu device is used for inference.
         """
         super().__init__()
 
-        devices, _ = initialize_device_settings(use_cuda=use_gpu, multi_gpu=False)
+        resolved_devices, _ = initialize_device_settings(devices=devices, use_cuda=use_gpu, multi_gpu=False)
         self.model = _Text2SpeechModel.from_pretrained(
-            model_name_or_path, device=devices[0].type, **(transformers_params or {})
+            model_name_or_path, device=resolved_devices[0].type, **(transformers_params or {})
```

Comment:
From what I understand here it looks like only one device can be used to load the `_Text2SpeechModel`. If that is True should we enforce that `devices` can only have a length of `1` using an assert statement? Otherwise users might be confused as to why only one device is used if they provide multiple.

Reply:
Thinking more on this I wonder if it would be better to change the parameter from `devices` to `device` and make it type `Union[str, torch.device]` (so removing the List aspect altogether). Then we could just make sure to pass `device` as a list to `initialize_device_settings()`. For example, 
```python
resolved_devices, _ = initialize_device_settings(devices=[device], use_cuda=use_gpu, multi_gpu=False)
```
What do you think?

Reply:
I thought about this as well. But is it possible that in the future there'll be support for multiple devices for components? The best trade-off seems to be to allow a list but to log a warning that if multiple devices are passed the first on is selected. We can do this in all cases where only one device is currently supported.  Thoughts?

Reply:
Yeah, that sounds good to me. I like the idea of just logging a warning, but having the code still run.